python/lsst/ts/atbuilding/csc/building_csc.py

```diff
@@ -0,0 +1,378 @@
+    async def do_openVentGate(self, data):
+        """Implement the ``openVentGate`` command."""
+        self.assert_enabled()
+        args = " ".join([str(i) for i in data.gate])
+        await self.run_command(f"open_vent_gate {args}")
+
+    async def do_resetExtractionFanDrive(self, data):
+        """Implement the ``resetExtractionFanDrive`` command."""
+        self.assert_enabled()
+        await self.run_command("reset_extraction_fan_drive")
+
+    async def do_setExtractionFanDriveFreq(self, data):
+        """Implement the ``setExtractionFanDriveFreq`` command."""
+        self.assert_enabled()
+        await self.run_command(f"set_extraction_fan_drive_freq {data.targetFrequency}")
+
+    async def do_setExtractionFanManualControlMode(self, data):
+        """Implement the ``setExtractionFanControlMode`` command."""
+        self.assert_enabled()
+        await self.run_command(
+            f"set_extraction_fan_manual_control_mode {data.enableManualControlMode}"
+        )
+
+    async def do_startExtractionFan(self, data):
+        """Implement the ``startExtractionFan`` command."""
+        self.assert_enabled()
+        await self.run_command("start_extraction_fan")
+
+    async def do_stopExtractionFan(self, data):
+        """Implement the ``stopExtractionFan`` command."""
+        self.assert_enabled()
+        await self.run_command("stop_extraction_fan")
+
+    async def run_command(self, command: str) -> None:
+        """Sends a command to the RPi. It writes to the TCP port,
+        and then monitors an `asyncio.Queue`. The response is
+        written to the queue by the `listen_for_messages` method.
+        """
+        await asyncio.wait_for(
+            self.client.write_str(command + "\r\n"), timeout=TCP_TIMEOUT
+        )
+
+        # Wait for a response
+        command_name = command.split()[0]
+        response = await asyncio.wait_for(
+            self.response_queue[command_name].get(), timeout=TCP_TIMEOUT
+        )
+        if response["error"] != 0:
+            # If an error code is supplied, log the error and
+            # raise an exception.
+            self.log.error(
+                "Error response received from command: "
+                + command
+                + " --> "
+                + json.dumps(response)
+            )
+            raise salobj.ExpectedError(json.dumps(response))
+
+        return response
+
+    async def listen_for_messages(self):
+        """Receives messages from the RPi. If the message contains an event
+        (command starts with "evt_") or telemetry (command is "telemetry") it
+        passes the message to the appropriate handler. Otherwise, it sends it
+        to the queue for that command, to be handled by the method that
+        called that command.
+        """
+        while self.client.connected:
+            try:
+                # Receive a message and format it as JSON.
+                self.listen_task = asyncio.create_task(self.client.read_str())
```

Comment:
I don't think you need to make all this. Consider simplifying it to something like:

```
message = await asyncio.wait_for(
    self.client.read_str(),
    timeout=SERVER_MESSAGE_TIMEOUT,
)
```

Reply:
I think it needs a task so that it can be cancelled in close(). I've taken away the timeout though.

Reply:
Ah, interesting... I have not seen that...